sources/src/ChatGPTPopover.tsx

```diff
@@ -1,79 +1,141 @@
-import {
-  AppBar,
-  AppBarProps,
-  Backdrop,
-  IconButton,
-  paperClasses,
-  Popover,
-  PopoverProps,
-  Typography
-} from '@mui/material';
-import CloseRounded from '@mui/icons-material/CloseRounded';
+import { AppBarProps, paperClasses, Popover, PopoverProps, useTheme } from '@mui/material';
 import ChatGPT, { ChatGPTProps } from './ChatGPT.tsx';
-import React from 'react';
+import React, { useCallback, useEffect, useState } from 'react';
+import MinimizedBar from '@craftercms/studio-ui/components/MinimizedBar';
+import DialogHeader from '@craftercms/studio-ui/components/DialogHeader/DialogHeader';
+import AlertDialog from '@craftercms/studio-ui/components/AlertDialog';
+import PrimaryButton from '@craftercms/studio-ui/components/PrimaryButton/PrimaryButton';
+import SecondaryButton from '@craftercms/studio-ui/components/SecondaryButton/SecondaryButton';
 
 export interface ChatGPTPopoverProps extends PopoverProps {
   appBarTitle?: string;
   width?: number | string;
   height?: number | string;
   appBarProps?: AppBarProps;
-  chatGPTProps: ChatGPTProps;
+  chatGPTProps?: ChatGPTProps;
+  isMinimized?: boolean;
+  onMinimize?: () => void;
+  onMaximize?: () => void;
 }
 
-function ChatGPTPopover(props: ChatGPTPopoverProps) {
+function ChatGPTPopover(props: Readonly<ChatGPTPopoverProps>) {
+  const theme = useTheme();
   const {
     open,
     onClose,
-    appBarProps,
     chatGPTProps,
+    isMinimized = false,
+    onMinimize,
+    onMaximize,
     appBarTitle = 'AI Assistant',
     width = 450,
     height = 500,
     ...popoverProps
   } = props;
+
+  const [openAlertDialog, setOpenAlertDialog] = useState(false);
+
+  const handleMinimize = useCallback(() => {
+    onMinimize?.();
+  }, [onMinimize]);
+
+  useEffect(() => {
+    const handleKeyDown = (event: KeyboardEvent) => {
+      if (event.key === 'Escape' && !isMinimized) {
+        setOpenAlertDialog(true);
+      }
+    };
+
+    if (open) {
+      document.addEventListener('keydown', handleKeyDown);
+    }
+
+    return () => {
+      document.removeEventListener('keydown', handleKeyDown);
+    };
+  }, [open, isMinimized, handleMinimize]);
```

Comment:
The close trigger for backdrop and escape are handled more robustly by the Popover component. Use the Popover onClose instead of manually listening for it. Inside your handleClose, you'd call the onClose prop or open the alert dialog if needed.

Also, don't think you need the useCallback for the handleMinimize anymore as the effect goes away.

Reply:
Updated to open confirm alert onClose